dbt_coves/tasks/blue_green/main.py

```diff
@@ -0,0 +1,228 @@
+            self.production_database,
+            self.staging_database,
+            self.con,
+        )
+        self._check_and_drop_staging_db()
+
+        try:
+            # create staging db
+            self.cdb.create_database(self.staging_database)
+            # clones schemas and schema grants from production to pre_production
+            self.cdb.clone_database_schemas(self.production_database, self.staging_database)
+            # run dbt build
+            dbt_build_command: list = self.get_dbt_build_command()
+            console.print("Running dbt build")
+            RunDbtTask(self.args, self.coves_config).run_dbt(command=dbt_build_command)
+            # copy db grants from production db
+            self.cdb.clone_database_grants(self.production_database, self.staging_database)
+            # Swaps databases: Snowflake sql `alter database {blue} swap with {green}`
+            self._swap_databases()
+            # drops pre_production (ex production)
+            self.cdb.drop_database()
+        except Exception as e:
+            if self.get_config_value("drop_staging_db_on_failure"):
+                self.cdb.drop_database()
+            raise e
+
+        return 0
+
+    def get_dbt_build_command(self):
+        """
+        Returns the dbt build command to be run.
+        """
+        dbt_selector = self.get_config_value("dbt_selector") or []  # []
+        is_deferral = self.get_config_value("defer")
+        dbt_build_command = ["build", "--fail-fast"]
+        if is_deferral:
+            dbt_build_command.extend(["--defer", "--state", "logs", "-s", "state:modified+"])
+        else:
+            dbt_build_command.extend(dbt_selector)
+        if self.get_config_value("full_refresh"):
+            dbt_build_command.append("--full-refresh")
+        if self.args.target:
+            dbt_build_command.extend(["-t", self.args.target])
+        return dbt_build_command
+
+    def _swap_databases(self):
+        """
+        Swaps databases: Snowflake sql `alter database {blue} swap with {green}`
+        """
+        console.print("Swapping databases")
+        try:
+            sql = f"alter database {self.production_database} swap with {self.staging_database};"
+            self.con.cursor().execute(sql)
+        except Exception as e:
+            print(f"Error swapping databases: {e}")
+            raise e
+
+    def _check_and_drop_staging_db(self):
+        """
+        Checks if the staging database exists and drops it if it does.
+
+        Returns:
+            None
+        """
+        green_exists = self._check_if_database_exists()
+        if green_exists and self.drop_staging_db:
+            if self.drop_staging_db_after:
+                console.print(
+                    f"Green database {self.staging_database} exists."
+                    f"Waiting {self.drop_staging_db_after} minutes before dropping it"
+                )
+                for i in range(self.drop_staging_db_after):
```

Comment:
Shouldn't this raise an exception if the number of loops ended and the database still exists?

Reply:
Agree, if you explicitly set it to drop, and give it X minutes to try and drop, something might've happened and blue-green cannot continue

Done.